test/fn/edit-contacts.spec.js

```diff
@@ -0,0 +1,174 @@
+                countFilesInDir(expectedDocsDirOneCol),
+                `Different number of files in ${saveDocsDir} and ${expectedDocsDirOneCol}.`);
+    compareDocuments(expectedDocsDirOneCol);
+  });
+  
+  it(`should add nested columns to the json docs perfectly`, async function(){
+
+    const parseResult = {
+      colNames: [],
+      csvFiles: ['contact.nested.csv'],
+      docDirectoryPath: path.resolve(environment.pathToProject, 'json_docs'),
+    };
+    const extraArgs = sinon.stub();
+    extraArgs.returns(parseResult);
+    editContactsModule.__set__('parseExtraArgs', extraArgs);
+
+    await editContactsModule.execute();
+
+    assert.equal(countFilesInDir(saveDocsDir),
+                countFilesInDir(expectedDocsDirNested),
+                `Different number of files in ${saveDocsDir} and ${expectedDocsDirOneCol}.`);
+    compareDocuments(expectedDocsDirNested);
+  });
+
+  it(`should fail when wrong column names are provided`, async function(){
+
+    const parseResult = {
+      colNames: ['enmch'],
+      csvFiles: ['contact.csv'],
+      docDirectoryPath: path.resolve(environment.pathToProject, 'json_docs'),
+    };
+    const extraArgs = sinon.stub();
+    extraArgs.returns(parseResult);
+    editContactsModule.__set__('parseExtraArgs', extraArgs);
+    
+    try {
+      await editContactsModule.execute();
+      assert.fail('should throw an error when wrong column names are provided');
+    } catch (err) {
+      expect(err.message).to.be.equal('The column name(s) specified do not exist.');
+    }
+  }); 
+
+  it(`should fail when protected column names are provided`, async function(){
+
+    const parseResult = {
+      colNames: ['parent'],
+      csvFiles: ['contact.test.csv'],
+      docDirectoryPath: path.resolve(environment.pathToProject, 'json_docs'),
+    };
+    const extraArgs = sinon.stub();
+    extraArgs.returns(parseResult);
+    editContactsModule.__set__('parseExtraArgs', extraArgs);
+    
+    try {
+      await editContactsModule.execute();
+      assert.fail('should throw an error when protected names are provided');
+    } catch (err) {
+      expect(err.message).to.include('this property name is protected.');
+    }
+  }); 
+
+  it(`should fail when DB doesn't contain the requested _id's`, async function(){
+    
+    const getIDs = sinon.stub();
+    getIDs.returns(['wrongDocumentID']);
+    editContactsModule.__set__('getIDs', getIDs);
+    try {
+      await editContactsModule.execute();
+      assert.fail('should throw an error when requested ID cannot be found on the database');
+    } catch (err) {
+      expect(err.message).to.include('could not be found.');
+    }
+  });
+
+  it(`should fail when document type is not a contact`, async function(){
+
+    await pouch.put({
+      _id: 'documentID',
+      type: 'data_record'
+    });
+    const getIDs = sinon.stub();
+    getIDs.returns(['documentID']);
+    editContactsModule.__set__('getIDs', getIDs);
```

Comment:
Please don't `__set__` `getIDs` here. Rather, setup your database and test csv so your module tries to edit an inexistent doc. Same goes for the other test.